libgnucash/engine/Transaction.c

```diff
@@ -3100,5 +3100,106 @@ _utest_trans_fill_functions (void)
     return func;
 }
 
+static void
+record_price(Split *split,
+             PriceSource source)
+{
+    Transaction *trans;
+    Account *account;
+    QofBook* book;
+    GNCPriceDB* pricedb;
+    gnc_commodity* comm;
+    gnc_commodity* curr;
+    GNCPrice* price;
+    gnc_numeric price_value, value, amount;
+    int scu;
+    time64 time;
+    gboolean swap;
+
+    account = xaccSplitGetAccount(split);
+    if (!xaccAccountIsPriced(account)) {
+       return;
+    }
+    amount = xaccSplitGetAmount(split);
+    if (gnc_numeric_zero_p(amount)) {
+       return;
+    }
+    trans = xaccSplitGetParent(split);
+    value = gnc_numeric_div(xaccSplitGetValue(split), amount,
+                            GNC_DENOM_AUTO,
+                            GNC_HOW_DENOM_EXACT);
+    book = qof_instance_get_book (QOF_INSTANCE (account));
+    pricedb = gnc_pricedb_get_db (book);
+    comm = xaccAccountGetCommodity (account);
+    curr = xaccTransGetCurrency (trans);
+    scu = gnc_commodity_get_fraction (curr);
+    swap = FALSE;
+    time = xaccTransGetDate(trans);
+    price = gnc_pricedb_lookup_day_t64 (pricedb, comm, curr, time);
+    if (gnc_commodity_equiv (comm, gnc_price_get_currency (price)))
+        swap = TRUE;
+
+    if (price)
+    {
+        price_value = gnc_price_get_value (price);
+        if (gnc_numeric_equal (swap ? gnc_numeric_invert (value) : value,
+                               price_value))
+        {
+            gnc_price_unref (price);
+            return;
+        }
+        if (gnc_price_get_source (price) < PRICE_SOURCE_XFER_DLG_VAL)
```

Comment:
I'd think better compare with individual enums instead of relying that the existing price source is less than a threshold?

Reply:
If I understand the code correctly, these enums are in order of preference.  So if the existing price source is preferred over this imported source,  we skip the update.

Reply:
@dl-g is correct, it's a precedence check.